system/lib/wasmfs/wasmfs.cpp

```diff
@@ -103,6 +104,43 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+long __syscall_fstat64(long fd, long buf) {
+  // Release FileTable lock after accessing desired open file.
+  auto currentOpenFile = [&fd] {
```

Comment:
Do we need to keep the file descriptor locked for the duration of this function or just the underlying file?

Reply:
That's true, I can modify the lambda to just return the file handle instead.